pip/compat.py

```diff
@@ -41,7 +46,11 @@ def console_to_str(s):
         try:
             return s.decode(sys.__stdout__.encoding)
         except UnicodeDecodeError:
-            return s.decode('utf_8')
+            if WINDOWS:
+                from ctypes import cdll
+                return s.decode("cp" + str(cdll.kernel32.GetACP()), 'replace')
```

Comment:
I've check the subprocess use locale.getpreferredencoding() as its default encoding.
So I follow the code to [here](https://github.com/python/cpython/blob/ad4a0cc519a5bb14204324e0e32e976318f9e6ce/Modules/_localemodule.c#L299),  and [here](https://github.com/python/cpython/blob/0710d754255e731e6fcc3f206b51db6156da17c8/Python/pylifecycle.c#L231),  it seems python use GetACP instead of GetOEMCP on Windows.
Maybe better way to solve this problem, is try to change the console_to_str function directly to

```return s.decode(locale.getpreferredencoding(), 'replace')```  no matter if it's windows or not?